api-server/server.js

```diff
@@ -124,6 +140,42 @@ app.post("/authenticate", (req, res) => {
   });
 });
 
+
+/******************************* Rooms ********************************/
+
+app.get("/rooms", passport.authenticate(["jwt"], { session: false }), (req, res) => {
+  let userLatitude = req.user.latitude;
+  let userLongitude = req.user.longitude;
+
+  let nearbyRoomPromise = Room.query().select(
+    raw("*, (6371 * acos(cos(radians(:userLatitude)) * " + 
```

Comment:
dat trigo tho. 6371 magic number? and `acos(cos(` is to get degrees ah. My trigo knowledge has floated away for quite a while lol.

Reply:
Earth radius in kilometres